tools/syz-declextract/syz-declextract.cpp

```diff
@@ -1,279 +1,103 @@
-      "CryptFlushTimeValidObject",
-      "CryptProtectDataNoUI",
-      "CryptUnprotectDataNoUI",
-      "NsServerBindSearch",
-      "NsClientBindSearch",
-      "NsClientBindDone",
-      "GetOpenCardNameA",
-      "SubscribeServiceChangeNotifications",
-      "UnsubscribeServiceChangeNotifications",
-      "GetThreadDescription",
-      "SetThreadDescription",
-      "DialogControlDpi",
-      "SetDialogDpiChangeBehavior",
-      "GetDialogDpiChangeBehavior",
-      "RpcServer",
-      "DecodePointer",
-      "DecodeRemotePointer",
-      "DecodeSystemPointer",
-      "EncodePointer",
-      "EncodeRemotePointer",
-      "EncodeSystemPointer",
-      "UnmapViewOfFile2",
-      "MapViewOfFileNuma2",
-      "DeriveCapabilitySidsFromName",
-      "QueryAuxiliaryCounterFrequency",
-      "ConvertPerformanceCounterToAuxiliaryCounter",
-      "ConvertAuxiliaryCounterToPerformanceCounter",
-      "FreePropVariantArray",
-      "PropVariantCopy",
-      "PropVariantClear",
-      "InitiateShutdown",
-      "ExitWindowsEx",
-      "LockWorkStation",
-      "InitiateSystemShutdown",
-      "InitiateSystemShutdownEx",
-      "shutdown",
-    };
-    for (auto func: ignore_functions) {
-      if (strstr(fn.c_str(), func)) return true;
-    }
-    // These are already described:
-    const char *ignore_exact[] {
-      "CreateFileA",
-      "CloseHandle",
-      "VirtualAlloc",
-    };
-    for (auto func: ignore_exact) {
-      if (strcmp(fn.c_str(), func) == 0) return true;
+
+    if (!varDecl->getInit()) {
+      return;
     }
-    const char *ignore_files[] {
-      "/um/ole",
-      "htiface.h",
-      "objbase.h",
-      "HLink.h",
-      "urlmon.h",
-      "HlGuids.h",
-      "unknwn.h",
-      "unknwnbase.h",
-      "coguid.h",
-      "MsHtmHst.h",
-      "msime.h",
-      "ComSvcs.h",
-      "combaseapi.h",
-      "WbemGlue.h",
-      "OCIdl.h",
-      "mfapi.h",
-      "CompPkgSup.h",
-      "ole2.h",
-      "Ole2.h",
-      "oleidl.h",
-      "ObjIdl.h",
-      "WabDefs.h",
-      "objidl.h",
-    };
-    auto src = D->getSourceRange().getBegin().printToString(Context.getSourceManager());
-    if (strstr(src.c_str(), "/um/") == 0) return true;
-    for (auto file: ignore_files) {
-      if (strstr(src.c_str(), file)) return true;
+
+    auto values = llvm::dyn_cast<InitListExpr>(varDecl->getInit())->inits();
+    if (values.empty()) {
+      return;
     }
-    for (const ParmVarDecl *P : D->parameters()) {
-      auto typ = convertType(Context, P->getType());
-      if (typ == "") {
-        llvm::outs() << D->getNameInfo().getAsString() << ": UNKNOWN TYPE: " <<
-            QualType(P->getType()).getAsString() << "\n";
-        return true;
+    int argc =
+        *values[2]->getIntegerConstantExpr(*context).value().getRawData();
+    std::vector<std::string> arg_types(argc);
+    std::vector<std::string> args(argc);
+
+    if (argc) {
+      int i = 0;
+      for (const auto *item :
+           llvm::dyn_cast<InitListExpr>(
```

Comment:
I think there are too many newlines in this statement. What does clang-format say?